skops/card/tests/test_card.py

```diff
@@ -1593,3 +1593,26 @@ def test_visibility_with_card_save(self, card):
             "Jane Doe"
         )
         assert loaded.strip() == expected
+
+
+class TestCardTableOfContents:
+    @pytest.fixture
+    def card(self):
+        model = LinearRegression()
+        card = Card(model=model)
+        return card
+
+    def test_toc(self, card):
+        card.add_model_plot()
+        card.add_hyperparams()
+        card.add_metrics(accuracy=0.1)
+        card.add_get_started_code()
+        toc = card.create_toc()
+        exptected_toc = (
```

Comment:
This test would be easier to read if the expected result were not concatenated like this. E.g. something like this:

```
expected_toc = [
    "- Model description",
    "  - Intended uses & limitations,
    ...,
]
assert toc == "\n".join(expected_toc)
```

As is, it would be quite hard to spot if a section is missing for instance.

Also, could you please add a custom section with `model_card.add(...)` and delete an existing section with `model_card.delete(...)`, just to be sure?